### ISSUE TITLE
Exposing Stan functions errors on overloaded functions

### ISSUE THREAD

**Issue by gowerc** [stan-dev/cmdstanr#739]
**Describe the bug**
Whilst trying to use the functionality implemented in #702 I noticed that if you try to expose overloaded functions it will fail to compile

**To Reproduce**

`stanfile.stan` file:
```stan
functions {
    // Vectorized ifelse() similar as in R.
    row_vector ifelse(array[] int condition, row_vector yes, row_vector no) {
        row_vector[num_elements(yes)] result;
        for (i in 1:num_elements(yes)) {
            result[i] = condition[i] ? yes[i] : no[i];
        }
        return result;
    }

    vector ifelse(array[] int condition, vector yes, vector no) {
        return ifelse(condition, yes', no')';
    }
}
data { real y_mean; }
parameters { real y; }
model { y ~normal(y_mean, 1); }
```

Corresponding R calling program:

```R
library(cmdstanr)
mod <- cmdstan_model("stanfile.stan")
fit <- mod$sample(data = list(y_mean = 0), refresh = 0)
fit$expose_functions()
```

Which results in:

```
file82dd17068fa.cpp:185:17: error: redefinition of 'sourceCpp_5_ifelse'
RcppExport SEXP sourceCpp_5_ifelse(SEXP conditionSEXP, SEXP yesSEXP, SEXP noSEXP) {
                ^
file82dd17068fa.cpp:172:17: note: previous definition is here
RcppExport SEXP sourceCpp_5_ifelse(SEXP conditionSEXP, SEXP yesSEXP, SEXP noSEXP) {
```

**Expected behavior**
Honestly I'm not sure. Would be nice if there was a solution for this but I'm not sure one is even possible. Maybe documentation that says this functionality can't work with overloaded functions ? 

**Operating system**
Mac OS 12.6.1

**CmdStanR version number**
‘0.5.3’ (though installed via `devtools::install_git()` i.e. the latest github commit as of today)

**Additional context**
nope

**Comment by rok-cesnovar** [comment]
Yeah, this one will be very difficult to address. The problem is that Rcpp treats both `row_vector` and `vector` as a `SEXP` object in C++, so both function signatures are exactly the same in C++.

I didn't try, but I guess it would work if the functions had a different number of arguments. And probably even if yes/no args in your case would be arrays. I think documenting this shortcoming is all we can do at this point.

Thanks for the report!

**Comment by andrjohns** [comment]
Closed by #743
